internal/environmentscleanup/service.go

```diff
@@ -292,30 +284,53 @@ func (s *Service) triggerEnvironmentDeprovisioning(instance internal.Instance) e
 	return nil
 }
 
-func (s *Service) cleanUpRuntimeCRs(runtimesToDelete []runtime, kebInstancesToDelete []internal.Instance) *multierror.Error {
-	kebInstanceExists := func(runtimeID string) bool {
-		for _, instance := range kebInstancesToDelete {
-			if instance.RuntimeID == runtimeID {
-				return true
-			}
-		}
+func (s *Service) cleanUpRuntimeCRs(runtimeCRs []runtime) error {
+	s.logger.Infof("RuntimeCRs to process: %+v", runtimeCRs)
 
-		return false
+	if len(runtimeCRs) == 0 {
+		return nil
 	}
 
 	var result *multierror.Error
 
-	for _, runtime := range runtimesToDelete {
-		if !kebInstanceExists(runtime.ID) {
-			s.logger.Infof("Deleting runtime CR for runtimeID ID %q", runtime.ID)
-			err := s.deleteRuntimeCR(runtime)
-			if err != nil {
-				result = multierror.Append(result, err)
+	for _, runtime := range runtimeCRs {
+		s.logger.Infof("Deleting runtime CR for runtimeID ID %q and shoot name %q", runtime.ID, runtime.ShootName)
+		err := s.deleteRuntimeCR(runtime)
+		if err != nil {
+			result = multierror.Append(result, err)
+		}
+	}
+
+	if result != nil {
+		result.ErrorFormat = func(i []error) string {
+			var s []string
+			for _, v := range i {
+				s = append(s, v.Error())
 			}
+			return strings.Join(s, ", ")
 		}
 	}
 
-	return result
+	return result.ErrorOrNil()
+}
+
+func (s *Service) getRuntimeCR(shootName string) (runtime, error) {
```

Comment:
We fetch all Runtime CRs just to select one.
See comment in the place of invocation.